menas/ui/components/dataset/datasetDetail.controller.js

```diff
@@ -206,15 +208,21 @@ sap.ui.define([
 
     routeMatched: function (oParams) {
       if (Prop.get(oParams, "id") === undefined) {
-        this._datasetService.getTop()
+        this._datasetService.getTop().then(() => this.load())
       } else if (Prop.get(oParams, "version") === undefined) {
-        this._datasetService.getLatestByName(oParams.id)
+        this._datasetService.getLatestByName(oParams.id).then(() => this.load())
       } else {
-        this._datasetService.getByNameAndVersion(oParams.id, oParams.version)
+        this._datasetService.getByNameAndVersion(oParams.id, oParams.version).then(() => this.load())
       }
       this.byId("datasetIconTabBar").setSelectedKey("info");
     },
 
+    load: function() {
+      let current = this._model.getProperty("/currentDataset");
```

Comment:
This variable is somewhere called current and somewhere dataset. Having it named the same way might improve readability, but generally, I don't mind keeping it as is, because they have quite the small scope.